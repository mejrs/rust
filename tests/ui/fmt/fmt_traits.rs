use std::fmt::*;

macro_rules! impl_for{
    ($this:ident, $($trait:ident),*) => {
        $(
            impl std::fmt::$trait for $this {
                fn fmt(&self, _f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
                    todo!()
                }
            }
        )*
    }
}

pub struct A;
impl_for!(A, Debug, Pointer, Display);

fn main(){
    let a = A;
    format!("the thing: {a:X}");
    //~^ ERROR cannot format `A` with format specifier `X` [E0277]
}