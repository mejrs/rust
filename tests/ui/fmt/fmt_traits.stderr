error[E0277]: cannot format `A` with format specifier `X`
  --> $DIR/fmt_traits.rs:20:25
   |
LL |     format!("the thing: {a:X}");
   |                         ^^^^^
   |
   = note: the format specifier `X` forwards to trait `UpperHex`, but `A` does not implement it
   = help: `A` implements the following formatting traits:
           `Display`, which can be invoked by omitting the format specifier
           `Debug`, which can be invoked with the `?` format specifier
           `Pointer`, which can be invoked with the `p` format specifier
           

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0277`.
